paddle/gserver/layers/Projection.h

```diff
@@ -69,6 +69,17 @@ class Projection {
     forward();
   }
 
+  void forward(const Argument* in,
```

Comment:
Do not change this file.
We can add a PoolMaxWithMaskLayer, and create this layer in the PoolLayer::create when pool == "max-pool-with-mask" .